src/config/loading.rs

```diff
@@ -251,3 +398,28 @@ pub fn load(
 
     format::deserialize(&with_vars, format).map(|builder| (builder, warnings))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::load_builder_from_paths;
+    use crate::config::{ComponentKey, ConfigPath};
+    use std::path::PathBuf;
+
+    #[test]
```

Comment:
[nit] Can we also test a negative scenario here - e.g. an unknown component key?